lib/Report/TimeDisconnectedSummary.php

```diff
@@ -320,79 +394,47 @@ public function getResults(SanitizerInterface $sanitizedParams)
                     }
                 }
 
-                $body .= " ) ";
+                $displayBody .= ' ) ';
             }
         }
 
         if ($onlyLoggedIn) {
-            $body .= ' AND `display`.loggedIn = 1 ';
+            $displayBody .= ' AND `display`.loggedIn = 1 ';
         }
 
-        $body .= '
+        $displayBody .= '
             GROUP BY display.display, display.displayId
         ';
 
-        $filter = [
-            'tags' => $sanitizedParams->getString('tags'),
-            'onlyLoggedIn' => $sanitizedParams->getCheckbox('onlyLoggedIn') == 1,
-            'exactTags' => $sanitizedParams->getCheckbox('exactTags')
-        ];
-
         // Sorting?
-        $filterBy = $this->gridRenderFilter($filter);
         $sortOrder = $this->gridRenderSort($sanitizedParams);
 
         $order = '';
         if (is_array($sortOrder)) {
             $order .= 'ORDER BY ' . implode(',', $sortOrder);
         }
 
-        $limit = '';
-
-        // Paging
-        $filterBy = $this->sanitizer->getSanitizer($filterBy);
-        if ($filterBy !== null && $filterBy->hasParam('start') && $filterBy->hasParam('length')) {
-            $limit = ' LIMIT ' . intval($filterBy->getInt('start', ['default' => 0])) . ', '
-                . $filterBy->getInt('length', ['default' => 10]);
-        }
-
-        $sql = $select . $body . $order . $limit;
-        $maxDuration = 0;
-        $rows = [];
-
-        foreach ($this->store->select($sql, $params) as $row) {
-            $maxDuration = $maxDuration + $this->sanitizer->getSanitizer($row)->getDouble('duration');
-        }
-
-        if ($maxDuration > 86400) {
-            $postUnits = __('Days');
-            $divisor = 86400;
-        } elseif ($maxDuration > 3600) {
-            $postUnits = __('Hours');
-            $divisor = 3600;
-        } else {
-            $postUnits = __('Minutes');
-            $divisor = 60;
-        }
+        // Get a list of displays by filters
+        $displaySql = $displaySelect . $displayBody . $order;
 
-        foreach ($this->store->select($sql, $params) as $row) {
-            $sanitizedRow = $this->sanitizer->getSanitizer($row);
-
-            $entry = [];
-            $entry['displayId'] = $sanitizedRow->getInt(('displayId'));
-            $entry['display'] = $sanitizedRow->getString(('display'));
-            $entry['timeDisconnected'] =  round($sanitizedRow->getDouble('duration') / $divisor, 2);
-            $entry['timeConnected'] =  round($sanitizedRow->getDouble('filter') / $divisor - $entry['timeDisconnected'], 2);
-            $entry['postUnits'] = $postUnits;
-
-            $rows[] = $entry;
+        // Displays by filter
+        $displays = [];
+        foreach ($this->store->select($displaySql, $params) as $displayRow) {
```

Comment:
Doing two loops means that the sorting is broken? 

You should only add to `$rows` in this loop, and always pull in the time disconnected/connected from the `$disconnectedDisplays` array.